Homework-06-documentation-testsanswers.ipynb

```diff
@@ -0,0 +1,201 @@
+      "  File \"<ipython-input-5-7a63d35ea9d3>\", line 28, in test_count_words\n",
+      "    stat=TextStats(text)\n",
+      "TypeError: __init__() takes 1 positional argument but 2 were given\n",
+      "\n",
+      "----------------------------------------------------------------------\n",
+      "Ran 2 tests in 0.004s\n",
+      "\n",
+      "FAILED (errors=1)\n"
+     ]
+    },
+    {
+     "ename": "TypeError",
+     "evalue": "__init__() takes 1 positional argument but 2 were given",
+     "output_type": "error",
+     "traceback": [
+      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
+      "\u001b[0;31mTypeError\u001b[0m                                 Traceback (most recent call last)",
+      "\u001b[0;32m<ipython-input-5-7a63d35ea9d3>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[1;32m     34\u001b[0m \u001b[0mtext\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m\"Betty Botter bought some butter. But she said the butter’s bitter. If I put it in my batter, it will make my batter bitter. But a bit of better butter will make my batter better. So it was better Betty Botter bought a bit of better butter.\"\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     35\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 36\u001b[0;31m \u001b[0mstat\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mTextStats\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mtext\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     37\u001b[0m \u001b[0mprint\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mstat\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mcount_words\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
+      "\u001b[0;31mTypeError\u001b[0m: __init__() takes 1 positional argument but 2 were given"
+     ]
+    }
+   ],
+   "source": [
+    "import pandas as pd\n",
+    "\n",
+    "class TextStats:\n",
+    "    \n",
+    "    def __init__(self):\n",
+    "        self.text = text\n",
+    "        \n",
+    "    def count_words(self):\n",
+    "        words = self.text.split(\" \")\n",
+    "        counts = {}\n",
+    "        for word in words:\n",
+    "            if word.endswith(\".\") or word.endswith(\",\"):\n",
+    "                word = word[:-1]\n",
+    "            word = word.lower()\n",
+    "            if word not in counts:\n",
+    "                counts[word] = 1\n",
+    "            else:\n",
+    "                counts[word] += 1\n",
+    "        return counts\n",
+    "    \n",
+    "import unittest\n",
+    "\n",
+    "class TestTextStats(unittest.TestCase):  \n",
+    "    def test__init(self):\n",
+    "        self.text=text\n",
+    "    \n",
+    "    def test_count_words(self):\n",
+    "        stat=TextStats(text)\n",
+    "        self.assertEqual(stat.count_words(\"butter\"), 2)\n",
```

Comment:
`count_words` doesn't take any arguments. And if you look at the code, you'll see that it returns a dictionary.